prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -83,19 +83,14 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
         context: ExecutionContext,
         actionHistory: ActionHistory,
     ): RetryItems? {
-        val logger: java.util.logging.Logger = context.logger
+        val logger: Logger = context.logger
 
         val restTransportInfo = transportType as RESTTransportType
         val reportId = "${header.reportFile.reportId}"
         val receiver = header.receiver ?: error("No receiver defined for report $reportId")
         val reportContent: ByteArray = header.content ?: error("No content for report $reportId")
-        // get the file name, or create one from the report ID, NY requires a file name in the POST
-        val fileName = Report.formFilename(
-            header.reportFile.reportId,
-            receiver.organizationName,
-            Report.Format.valueOf(receiver.translation.type),
-            header.reportFile.createdAt
-        )
+        // get the file name from blob url, or create one from the report metadata
+        val fileName = Report.formExternalFilename(header)
```

Comment:
@victor-chaparro and @oslynn : I updated RESTTransport to prefer to calculate the file name from the blob URL and if it does not exist it will essentially do the code above. Is this okay? Were there special requirements for the old name format for REST? I feel this change makes it more consistent with other transport protocols. Please see implementation of .formExternalFilename for more details on what that function exactly does. Also, see the updated RESTTTransportIntegrationTests to see how the filename would change now if the bodyurl was empty (which is almost never going to happen?)

Reply:
@oslynn I think there's a receiver you recently onboarded for ETOR that does use the old file format right? Will this break that

Reply:
The other transport protocols do what I have here, that is:
`val fileName = Report.formExternalFilename(header)`

I'm not sure we should be changing that logic per transport mechanism? Seems like this file name thing needs a greater refactor?